tests/test-backend-ops.cpp

```diff
@@ -3624,20 +3606,28 @@ static bool test_backend(ggml_backend_t backend, test_mode mode, const char * op
     test_cases.emplace_back(new test_falcon(2));
 #endif
 
-    // run tests
-    if (mode == MODE_GRAD) {
-        size_t n_ok = 0;
-        for (auto & test : test_cases) {
-            if (test->eval_grad(backend, op_name)) {
-                n_ok++;
+    return test_cases;
+}
+
+// Test cases for performance evaluation: should be representative of real-world use cases
+static std::vector<std::unique_ptr<test_case>> make_test_cases_perf() {
+    std::vector<std::unique_ptr<test_case>> test_cases;
+
+    for (int bs : {1, 8, 16, 32, 512}) {
```

Comment:
Matrix multiplications become more efficient for larger batch sizes. At the same time the number of runs is scaled in such a way that the total FLOPs are roughly constant. So you could consider removing 16 and instead adding 1024 or 2048. That would make the test faster and cover a wider range.

Reply:
I have changed it to include only 1 and 512. I think that should provide a good overview of the performance that can be expected during generation and prompt processing for people casually running the benchmark, but I expect that people working on optimizing an operation will modify this function to add the test cases relevant to what they are working on. It would be nice to be able to specify the parameters of the test cases from the command line as well, but that's a more complex change.

The way the number of runs was scaled was not very good, and I noticed that some of the test were too short and produced very inaccurate results. I changed it so that the memory size or flops is used as an initial estimate to determine how many times to duplicate the op in the graph, but the graph is evaluated repeatedly until it has run for least one second. This seem to produce much more reliable results.